src/Materials/Textures/Packer/packer.ts

```diff
@@ -0,0 +1,792 @@
+import { Engine } from "../../../Engines/engine";
+import { Constants } from "../../../Engines/constants";
+import { AbstractMesh } from "../../../Meshes/abstractMesh";
+import { VertexBuffer } from "../../../Meshes/buffer";
+import { Scene } from "../../../scene";
+import { Material } from "../../material";
+import { Texture } from "../texture";
+import { DynamicTexture } from "../dynamicTexture";
+import { Nullable } from "../../../types";
+import { Vector2 } from "../../../Maths/math.vector";
+import { Color3, Color4 } from "../../../Maths/math.color";
+import { TexturePackerFrame } from "./frame";
+
+/**
+* Defines the basic options interface of a TexturePacker
+*/
+export interface ITexturePackerOptions{
+
+    /**
+    * Custom targets for the channels of a texture packer.  Default is all the channels of the Standard Material
+    */
+    map?: string[];
+
+    /**
+    * the UV input targets, as a single value for all meshes. Defaults to VertexBuffer.UVKind
+    */
+    uvsIn?: string;
+
+    /**
+    * the UV output targets, as a single value for all meshes.  Defaults to VertexBuffer.UVKind
+    */
+    uvsOut?: string;
+
+    /**
+    * number representing the layout style. Defaults to LAYOUT_STRIP
+    */
+    layout?: number;
+
+    /**
+    * number of columns if using custom column count layout(2).  This defaults to 4.
+    */
+    colnum?: number;
+
+    /**
+	 * flag to update the input meshes to the new packed texture after compilation. Defaults to true.

```

Comment:
Still tabs issue in this file :(

Reply:
That is really weird my lint is not catching that....